src/Util/src/FMI3/fmi3_enums.c

```diff
@@ -148,64 +148,69 @@ const char* fmi3_initial_to_string(fmi3_initial_enu_t ini) {
-    /* continuous */{fmi3_initial_enu_unknown, fmi3_initial_enu_unknown,    fmi3_initial_enu_exact  , fmi3_initial_enu_calculated, fmi3_initial_enu_calculated, fmi3_initial_enu_unknown, fmi3_initial_enu_unknown}
+    /*              structural parameter       parameter                 calculated parameter,        input                     output                       local                        independent             */
+    /* constant */  {fmi3_initial_enu_unknown, fmi3_initial_enu_unknown, fmi3_initial_enu_unknown,    fmi3_initial_enu_unknown, fmi3_initial_enu_exact,      fmi3_initial_enu_exact,      fmi3_initial_enu_unknown},
+    /* fixed   */   {fmi3_initial_enu_exact,   fmi3_initial_enu_exact,   fmi3_initial_enu_calculated, fmi3_initial_enu_unknown, fmi3_initial_enu_unknown,    fmi3_initial_enu_calculated, fmi3_initial_enu_unknown},
+    /* tunable */   {fmi3_initial_enu_exact,   fmi3_initial_enu_exact,   fmi3_initial_enu_calculated, fmi3_initial_enu_unknown, fmi3_initial_enu_unknown,    fmi3_initial_enu_calculated, fmi3_initial_enu_unknown},
+    /* discrete */  {fmi3_initial_enu_unknown, fmi3_initial_enu_unknown, fmi3_initial_enu_unknown,    fmi3_initial_enu_exact  , fmi3_initial_enu_calculated, fmi3_initial_enu_calculated, fmi3_initial_enu_unknown},
+    /* continuous */{fmi3_initial_enu_unknown, fmi3_initial_enu_unknown, fmi3_initial_enu_unknown,    fmi3_initial_enu_exact  , fmi3_initial_enu_calculated, fmi3_initial_enu_calculated, fmi3_initial_enu_unknown}
 };
 
-FMILIB_EXPORT fmi3_variability_enu_t fmi3_get_default_valid_variability(fmi3_causality_enu_t c)
+fmi3_variability_enu_t fmi3_get_default_valid_variability(fmi3_causality_enu_t c, int isFloat)
 {
-    switch(c){
-    case fmi3_causality_enu_parameter:            return fmi3_variability_enu_fixed;
-    case fmi3_causality_enu_calculated_parameter: return fmi3_variability_enu_fixed;
-    case fmi3_causality_enu_input:                return fmi3_variability_enu_continuous;
-    case fmi3_causality_enu_output:               return fmi3_variability_enu_continuous;
-    case fmi3_causality_enu_local:                return fmi3_variability_enu_continuous;
-    case fmi3_causality_enu_independent:          return fmi3_variability_enu_continuous;
-    case fmi3_causality_enu_structural_parameter: return fmi3_variability_enu_fixed;
-    default:                                      return fmi3_variability_enu_unknown;
+    if (c >= fmi3_causality_enu_unknown) {
+        return fmi3_variability_enu_unknown;
+    }
+    // Spec: "The default for variables of causality `parameter`, `structural parameter` or `calculated parameter` is `fixed`"
+    if (c <= fmi3_causality_enu_calculated_parameter) {
+        return fmi3_variability_enu_fixed;
+    }
+
+    if (isFloat) {
+        // Spec: "The default for variables of type `<Float32>` and `<Float64>` and causality other than `parameter`, `structural parameter` or `calculated parameter` is <<continuous>>"
```

Comment:
Long line. Tip: Add this to your personal settings.json: `    "editor.rulers": [ 120 ],`

Reply:
Fixed.